solutions/factorial_calculator.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Module Name: factorial_calculator
+Description: This module provides a function to calculate the factorial of a given number.
+It uses a recursive approach to compute the factorial.
+
+The module includes: the following function:
+    - factorial_calculator(num): Returns the factorial of the input number num.
+
+Author: Linah Khayri
+Date Created: 29-12-2024
+"""
+
+
+def factorial_calculator(num: int | float) -> int:
+    """
+    This function calculates the factorial of a non-negative integer or whole float using recursion.
+
+    The factorial of a non-negative integer n is the product of all positive integers
+    less than or equal to num until we reach one.
+    For example, the factorial of num! = num * (num-1) * ... * 1.
+    Also 0 and 1 are special cases when it comes to calculating the factorial
+    as their factorial is 1
+
+    Parameters:
```

Comment:
this is to be applied to all ,  indentation on the comments like
Parameters:
          num:........